terraform/modules/rke2-cluster/templates/cloudinit-worker.yaml

```diff
@@ -19,6 +19,8 @@ write_files:
     server: https://${lb_address}:9345
     token: ${rke2_cluster_secret}
     cloud-provider-name: external
+    kubelet-arg:
```

Comment:
You also have to manually make this change on the running nodes. The cloudinit file only applies on new nodes when created by terraform. Existing nodes are untouched when that file changes. Because the node would need to be recreated and this have to be done a  bit more careful, at least on the control plane nodes.

So you can change this on `/etc/rancher/rke2/config.yaml` and restart `rke-server` on the control plane nodes and `rke-agent` on the worker only nodes.